lib/money/money.rb

```diff
@@ -603,6 +644,11 @@ def decimal_mark
 
   private
 
+  def dup_with_new_fractional(new_fractional, infinite_precision = nil)
```

Comment:
To make this method more useful, could you change it to `def dup_with(fractional: nil, currency: nil, bank: nil, infinite_precision: nil)`, defaulting every `nil` argument to its former value?

This is also a very good candidate for a separate PR against `master` to reduce the scope of this one

Reply:
Hmm, with this suggested change maybe the convenience conversion methods are even less necessary... if we made this method public that is.

Reply:
Well, convenience methods would be just shorthand wrappers for this method (and while Money objects are immutable - with little extra optimization - is not actually creating a new copy if the current object already has the right precision).